src/components/sidebarcom.jsx

```diff
@@ -1,133 +1,185 @@
-import React, { useState } from "react";
-import { FaBeer,FaPython } from 'react-icons/fa';
-import {DiCode, DiJavascript1,DiRust} from 'react-icons/di';
-import {GiBrokenHeart} from 'react-icons/gi';
-import {AiOutlineFileText,AiFillHeart} from 'react-icons/ai';
-import {SiCsharp} from 'react-icons/si';
-import CodeMirror from '@uiw/react-codemirror';
-import {okaidia} from '@uiw/codemirror-theme-okaidia';
-import {sublime} from '@uiw/codemirror-theme-sublime';
-import FileSaver, { saveAs } from 'file-saver';
+import React, { useEffect, useState } from "react";
+import { FaBeer, FaPython } from "react-icons/fa";
+import { DiCode, DiJavascript1, DiRust } from "react-icons/di";
+import { GiBrokenHeart } from "react-icons/gi";
+import { AiOutlineFileText, AiFillHeart } from "react-icons/ai";
+import { SiCsharp } from "react-icons/si";
+import CodeMirror from "@uiw/react-codemirror";
+import { okaidia } from "@uiw/codemirror-theme-okaidia";
+import { sublime } from "@uiw/codemirror-theme-sublime";
+import FileSaver, { saveAs } from "file-saver";
 // const writeFileP = require("write-file-p");
 
 function Sidebar(props) {
-    const { tabCount } = props;
-    const {clickblob,changestateblob} = props;
-    const {textingchan,settextingchan} = props;
-    let ouputext = '';
-    const [textin,setextin] = useState("");
-    const texting = document.querySelector('#textinput');
+  const {
+    tabCount,
+    themestate,
+    clickblob,
+    changestateblob,
+    textingchan,
+    settextingchan,
+  } = props;
 
-const [iconstate,seticonstate] = useState(<FaBeer className="w-5 h-5 absolute ml-3 pointer-events-none"/>);
-const {themestate} = props;
-console.log(themestate);
-const themefunc = () =>{
-    if(themestate === "okaidia"){
-        console.log("true");
-        return okaidia
-    }
-    else if (themestate === "sublime"){
-        return sublime
+  //   let ouputext = "";
+  const [textin, setextin] = useState("");
+  //   const texting = document.querySelector("#textinput");
+
+  const [iconstate, seticonstate] = useState(
+    <FaBeer className="w-5 h-5 absolute ml-3 pointer-events-none" />
+  );
+  const [theme, setTheme] = useState(() => {
+    if (themestate === "okaidia") {
+      return okaidia;
+    } else if (themestate === "sublime") {
+      return sublime;
     }
-}
+    return "";
+  });
 
-let counter = 0;
-const [compo, setCompo] = useState(<CodeMirror  options={{mode:'javascript', lineNumbers:true}} value="console.log('hello world')"
```

Comment:
I added this to state so I can loop through the multiple instances of it. So if the user clicks on the plus symbol another instance of codemirror will be created and we can have a counter managing each instance.

Reply:
for that you have to manage that as an array